webpush/encrypt.go

```diff
@@ -144,16 +168,25 @@ type EncryptionResult struct {
 //    - https://tools.ietf.org/html/draft-ietf-httpbis-encryption-encoding
 //    - https://en.wikipedia.org/wiki/Elliptic_curve_Diffie%E2%80%93Hellman
 //    - https://tools.ietf.org/html/draft-ietf-webpush-encryption
-func Encrypt(sub *Subscription, message string) (*EncryptionResult, error) {
+func Encrypt(sub *Subscription, message string, version Version) (*EncryptionResult, error) {
 	plaintext := []byte(message)
+
+	var maxPayloadLength int
+	if version == AESGCM {
+		maxPayloadLength = aesgcmMaxPayloadLength
+	} else {
+		maxPayloadLength = aes128gcmMaxPayloadLength
+	}
 	if len(plaintext) > maxPayloadLength {
 		return nil, fmt.Errorf("Payload is too large. The max number of bytes is %d, input is %d bytes.", maxPayloadLength, len(plaintext))
 	}
 
+	// sub.Key is the user agent's public key.
```

Comment:
This is the subscriptions public key rather then than the user agents key right? Also the naming makes this a little hard to parse, i.e. I'm assuming sub is subscription and Key is too vague to be meaningful in the context of an encryption flow. p256dh would be at least more obvious.